File: test/regression/regression_1.py
import os
from cmdstanpy import cmdstan_path, CmdStanModel
import numpy as np
import fileinput
import sys
# Run from command line: Python regression_1.py

alpha = 2
beta = 3
sigma = 5
N = 100
x = np.random.uniform(size = N)
y = np.random.normal(size = N, loc=alpha + beta * x, scale = sigma)

stan_data = {'N': N, 'x': x, 'y': y}

stan_program = CmdStanModel(stan_file='regression_1.stan')
stan_program.compile()
fit = stan_program.sample(data=stan_data,
                           csv_basename='./regression_1')


print(fit)

File: test/regression/regression_1.R
library("rstan")

alpha <- 2
beta <- 3
sigma <- 5
N <- 100
x <- runif(N, 0, 10)
y <- rnorm(N, alpha + beta*x, sigma)
stan_data <- list(N=N, x=x, y=y)

model <- stan_model("linear_regression_1.stan")
fit <- sampling(model, data=stan_data)
print(fit)

cat(
    "alpha =", alpha, "\n",
    "beta =", beta, "\n",
    "sigma =", sigma, "\n"
)